scipy/special/_logsumexp.py

```diff
@@ -125,3 +125,29 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
         return out, sgn
     else:
         return out
+
+
+def softmax(x, axis=None):
+    r"""
+    Softmax function
+
+    Compute the softmax of each element along an axis of X.
+
+    .. math:: \sigma(X)_j = \frac{e^{x_j}}{\sum_k e^{x_k}}
+
+    Parameters
+    ----------
+    x : array_like
+        Input array.
+    axis : int, optional
+           axis to compute values along. Default is None and softmax will be computed over all axes.
+
+    Returns
+    -------
+    s : ndarray
+        an array the same size as X. The result will sum to 1 along the specified axis.
```

Comment:
Could you add a `Notes` section with ``.. versionadded:: 1.1.0``, and an `Examples` section with a simple example?

Reply:
+1 for adding an `Examples` section; see https://github.com/scipy/scipy/issues/7168